pkg/sql/set_zone_config.go

```diff
@@ -962,6 +976,69 @@ func validateZoneAttrsAndLocalities(
 	return nil
 }
 
+// validateZoneLocalitiesForSecondaryTenants performs all the constraint/lease
+// preferences validation for secondary tenants. Secondary tenants are only
+// allowed to reference locality attributes as they only have access to region
+// information via the RegionServer. Even then, they're only allowed to
+// reference the "region" and "zone" tiers.
+//
+// Unlike the system tenant, we also catch typos in and validate prohibited
+// constraints. This is because secondary tenant must operate in the narrow
+// view exposed via the RegionServer and are not allowed to configure arbitrary
+// constraints (required or otherwise).
+func validateZoneLocalitiesForSecondaryTenants(
+	ctx context.Context, getRegions regionsGetter, zone *zonepb.ZoneConfig,
+) error {
+	toValidate := accumulateUniqueConstraints(zone)
+	resp, err := getRegions(ctx, &serverpb.RegionsRequest{})
+	if err != nil {
+		return nil
+	}
+	regions := make(map[string]struct{}, 0)
```

Comment:
nit: i would omit setting the size to 0 and let the default mechanism do the expanding. unless you expect it to be 0 most of the time (i'd think not if we're here)